src/dmd/dcast.d

```diff
@@ -839,6 +839,27 @@ MATCH implicitConvTo(Expression e, Type t)
             }
         }
 
+        override void visit(CatExp e)
+        {
+            enum LOG = false;
+            static if (LOG)
+            {
+                printf("CatExp::implicitConvTo(this=%s, type=%s, t=%s)\n", e.toChars(), e.type.toChars(), t.toChars());
+            }
+
+            visit(cast(Expression)e);
+            if (result != MATCH.nomatch)
+                return;
+
+            if (true)       // TODO: only if any indirections of e are immutable
```

Comment:
If element type of `e` has indirections they must be immutable to allow implicit conv to immutable. How do I check that?

Reply:
Solved.